data/transactions/logic/assembler.go

```diff
@@ -1101,6 +1101,63 @@ func typeDig(ops *OpStream, args []string) (StackTypes, StackTypes) {
 	return anys, returns
 }
 
+func typeCover(ops *OpStream, args []string) (StackTypes, StackTypes) {
+	if len(args) == 0 {
+		return oneAny, oneAny
+	}
+	n, err := strconv.ParseUint(args[0], 0, 64)
+	if err != nil {
+		return oneAny, oneAny
+	}
+	depth := int(n) + 1
+	anys := make(StackTypes, depth)
+	for i := range anys {
+		anys[i] = StackAny
+	}
+	returns := make(StackTypes, depth)
+	for i := range returns {
+		returns[i] = StackAny
+	}
+	idx := len(ops.typeStack) - depth
+	if idx >= 0 {
```

Comment:
Will the idx here ever be less than 0? Won't that be caught in the `opCover` function?

Reply:
I believe this check occurs first, as jj said below opCover should now be sure that it has the correct size stack

Reply:
This is actually because the arrays will create out of bounds error if you don't check here. It doesn't panic though